flax/training/common_utils.py

```diff
@@ -21,33 +25,80 @@
 
 
 def shard(xs):
+  """Helper for pmap to shard a pytree of arrays by local_device_count.
+
+  Args:
+    xs: a pytree of arrays.
+  Returns:
+    A matching pytree with arrays' leading dimensions sharded by the
+    local device count.
+  """
   local_device_count = jax.local_device_count()
   return jax.tree_util.tree_map(
       lambda x: x.reshape((local_device_count, -1) + x.shape[1:]), xs)
 
 
 def shard_prng_key(prng_key):
-  # PRNG keys can used at train time to drive stochastic modules
-  # e.g. DropOut. We would like a different PRNG key for each local
-  # device so that we end up with different random numbers on each one,
-  # hence we split our PRNG key and put the resulting keys into the batch
-  return jax.random.split(prng_key, num=jax.local_device_count())
+  """Helper to shard (aka split) a PRNGKey for use with pmap'd functions.
 
+   PRNG keys can used at train time to drive stochastic modules
+   e.g. Dropout. We would like a different PRNG key for each local
+   device so that we end up with different random numbers on each one,
+   hence we split our PRNG key.
 
-def onehot(labels, num_classes, on_value=1.0, off_value=0.0):
-  x = (labels[..., None] == jnp.arange(num_classes).reshape((1,) * labels.ndim + (-1,)))
-  x = lax.select(x, jnp.full(x.shape, on_value), jnp.full(x.shape, off_value))
-  return x.astype(jnp.float32)
+  Args:
+    prng_key: JAX PRNGKey
+  Returns:
+    A new array of PRNGKeys with leading dimension equal to local device count.
+  """
+  return jax.random.split(prng_key, num=jax.local_device_count())
 
 
 def stack_forest(forest):
+  """Helper function to stack the leaves of a sequence of pytrees.
+
+  Args:
+    forest: a sequence of pytrees (e.g tuple or list) of matching structure
+    whose leaves are arrays with individually matching shapes.
```

Comment:
Missing 2 spaces before "whose" to continue the doc for the current argument (`forest`) in a new line.
![Screenshot from 2022-08-04 19-32-22](https://user-images.githubusercontent.com/5862228/182977512-306cf404-1893-4993-8aea-ff8420ab5160.png)